snap/pack/pack.go

```diff
@@ -180,10 +180,24 @@ func excludesFile() (filename string, err error) {
 	return filename, err
 }
 
+type Options struct {
+	TargetDir, SnapName, Compression string
+}
+
 // Snap the given sourceDirectory and return the generated
 // snap file
-func Snap(sourceDir, targetDir, snapName string) (string, error) {
-	info, err := prepare(sourceDir, targetDir)
+func Snap(sourceDir string, opts *Options) (string, error) {
```

Comment:
Nitpick, I'd consider using `Snap(sourceDir string, opts Options)`, so that the call sites do not have bare `nil`s, but are more explicit instread, eg. `pack.Snap(snapSource, pack.Options{})`.

However we have a bunch of places where we pass bare `nil`s, so I won't mind if we do the same here. FWIW, we could also have a package level:

```
var Defaults = pack.Options{}
// or if we go with defaults
var Defaults *Options = nil
```
though with the 2nd form, it's still possible to pass bare `nil` as parameter.

Reply:
Thank you, added the second one now